dev/final_action.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include "cxx_polyfill.h"
+
+template<class F>
+class final_action {
+  public:
+    explicit final_action(F f) noexcept : f_{std::move(f)} {}
+
+    final_action(final_action&& other) noexcept :
+        f_{std::move(other.f_)}, invoke_{std::exchange(other.invoke_, false)} {}
+
+    final_action(const final_action&) = delete;
+    final_action& operator=(const final_action&) = delete;
+    final_action& operator=(final_action&&) = delete;
+
+    ~final_action() {
+        if(invoke_) {
+            f_();
+        }
+    }
+
+  private:
+    F f_;
+    bool invoke_ = true;
+};
+
+template<class F>
+auto finally(F&& f) noexcept {
```

Comment:
you use `finally` with lambda which throws. In such cases `std::terminate` gets called instead of allowing user to catch an exception. Library which may terminate the whole app by design is not what people want to use. Such code is not going to be merged into `sqlite_orm` ever.

Reply:
I removed this